### DISCUSSION TITLE
How Can I Inject MUI TextField and NumberFormat in Controller

### DISCUSSION THREAD

**Discussion by albertolotz** [react-hook-form/react-hook-form#3239] [General] [answered]
Look bellow.
I would like to use Controller to format a number, then I'm trying to use Controller as {NumberFormat}, but I lost a style TextField.

Do you know how Can I do it? 
The exemplo MUI home page doesn't work with react-hook-form.

Thank you
Ps: in my code TextField is commented ...

'''
              <TextField
                className={classes.spacingNone}
                size="small"
                fullWidth
                id="price"
                variant="outlined"
                margin="normal"
                label="Valor Mão de Obra - Área"
                name="price"
                inputRef={register}
                defaultValue=""
              />
              

              <Controller
                as={NumberFormat}
                thousandSeparator
                name="price"
                control={control}
              />
![Captura de tela de 2020-10-22 20-43-50](https://user-images.githubusercontent.com/55667164/96940902-526e8200-14a7-11eb-94ce-ba6faf30a816.png)
'''

**Comment by bluebill1049** [comment]
try to use `render` prop

**Reply by albertolotz** [reply]
now, de style work but de format not.... sorry my little experience, I'm learning...

'''<Controller
                as={TextField}
                className={classes.spacingNone}
                size="small"
                fullWidth
                id="price"
                variant="outlined"
                margin="normal"
                label="Valor Mão de Obra - Área"
                name="price"
                inputRef={register}
                defaultValue=""
                control={control}
                render={props => <NumberFormat thousandSeparator {...props} />}
              />'''

**Comment by bluebill1049** [comment, marked as answer]
<img width="690" alt="Screen Shot 2020-10-23 at 11 05 14 am" src="https://user-images.githubusercontent.com/10513364/96942010-a2892b00-151f-11eb-9eab-34a4d36eb10d.png">

Try above render prop and here is the examples: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by albertolotz** [reply]
Tks My friend, for you help!!! you is fast and generous.
To help the comunity , I solved.
'''
<Controller
                as={
                  <NumberFormat
                    customInput={TextField}
                    decimalSeparator=","
                    thousandSeparator="."
                    decimalScale={2}
                    autoComplete="off"
                  />
                }
                control={control}
                className={classes.spacingNone}
                size="small"
                fullWidth
                id="price"
                variant="outlined"
                margin="normal"
                label="Valor Mão de Obra - Área"
                name="price"
                defaultValue=""
              />
'''
![Captura de tela de 2020-10-22 21-18-42](https://user-images.githubusercontent.com/55667164/96942688-34575080-14ac-11eb-8b51-94d4bbbdeefc.png)

**Reply by bluebill1049** [reply]
my pleasure :) make sure you read the 'render' prop section as well.
